streams/src/main/java/org/apache/kafka/streams/processor/internals/RecordDeserializer.java

```diff
@@ -68,40 +68,49 @@ ConsumerRecord<Object, Object> deserialize(final ProcessorContext<?, ?> processo
                 Optional.empty()
             );
         } catch (final Exception deserializationException) {
-            final DeserializationExceptionHandler.DeserializationHandlerResponse response;
-            try {
-                response = deserializationExceptionHandler.handle(
-                    (InternalProcessorContext<?, ?>) processorContext,
-                    rawRecord,
-                    deserializationException);
-            } catch (final Exception fatalUserException) {
-                log.error(
-                    "Deserialization error callback failed after deserialization error for record {}",
-                    rawRecord,
-                    deserializationException);
-                throw new StreamsException("Fatal user code error in deserialization error callback", fatalUserException);
-            }
+            handleDeserializationFailure(deserializationExceptionHandler, processorContext, deserializationException, rawRecord, log, droppedRecordsSensor);
+            return null;
```

Comment:
Semantics of `return null` is unclear w/o looking into `handleDeserializationFailure` -- we should at least put a comment `//  'handleDeserializationFailure' would either throw or swallow -- if we swallow we need to skip the record by returning 'null'` (or something like this)?

Reply:
yeah we can add a note